CRM/Utils/System/Drupal8.php

```diff
@@ -100,6 +111,11 @@ public function createUser(&$params, $mail) {
         break;
     }
 
+    // If this is a user creating their own account, login them in!
+    if ($account->isActive() && $user->isAnonymous()) {
+      \user_login_finalize($account);
```

Comment:
In the Drupal 7 version of this module, it's submitting the Drupal user registration form which will call this as part of it's submit handle. In Drupal 8, since it's creating the Drupal user via the API and not submitting any form, we need to login the user in manually here, following the right rules per Drupal's settings.